src/pages/dev/gas-service/on-chain.mdx

```diff
@@ -0,0 +1,126 @@
+# On-Chain Gas Estimation (alpha)
+
+On-chain gas estimation allows you to estimate interchain transaction gas fees directly from a Solidity contract. You should only use this capability if you are unable to estimate gas off-chain. This on-chain API has been designed for specific use cases such as DApps and services that only integrate and have no UI for a user to initiate transactions. Most developers should instead use the [off-chain `estimateGas` API](/dev/axelarjs-sdk/axelar-query-api#estimategasfee)
+
+## Usage
+The on-chain API provided by the [IInterchainGasEstimation](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/interfaces/IInterchainGasEstimation.sol) interface which is implemented by the Axelar [Gas Service](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/interfaces/IAxelarGasService.sol) contract._createMdxContent
```

Comment:
I think Axelar [Gas Service] should be [Axelar Gas Service] since Axelar is included in the name of the contract itself.

Reply:
Also not sure what the `_createMdxContent` is in reference to here?

Reply:
Do you think we should make mention of the buffer we have set? Estimation might be high due to the buffer and devs might want to know why. Also devs might be wondering if we have a buffer at all

Reply:
updated link. The createMdx content was an invalid autocomplete I missed.